I tried using this with `build_type=Debug`, but it failed with (it builds fine for Release):

```
CMake Error at cmakedeps_macros.cmake:81 (message):
  Library 'box3d' not found in package.
```

The reason is that upstream sets `CMAKE_DEBUG_POSTFIX "d"` and applies it to the target, so the Debug lib is `libbox3dd.a`, but `package_info()` hardcodes `libs = ["box3d"]`.

Fixing it in `package_info()` works:

```python
debug_postfix = "d" if self.settings.build_type == "Debug" else ""
self.cpp_info.libs = [f"box3d{debug_postfix}"]
```

(Setting `CMAKE_DEBUG_POSTFIX=""` from the toolchain doesn't help since the plain `set()` in the subdir overrides it.)

--- a/recipes/box3d/all/conanfile.py
+++ b/recipes/box3d/all/conanfile.py
@@ -30,9 +30,7 @@ class Box3dConan(ConanFile):
     }
 
     implements = ["auto_shared_fpic"]
-
-    def validate(self):
-        check_min_cppstd(self, 17)
+    languages = "C"
 
     def build_requirements(self):
         self.tool_requires("cmake/[>=3.22]")
@@ -78,7 +76,10 @@ class Box3dConan(ConanFile):
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "box3d")
         self.cpp_info.set_property("cmake_target_name", "box3d::box3d")
-        self.cpp_info.libs = ["box3d"]
+        postfix = "d" if self.settings.build_type == "Debug" else ""
+        self.cpp_info.libs = [f"box3d{postfix}"]
+        if self.settings.os in ("Linux", "FreeBSD"):
+            self.cpp_info.system_libs.append("m")
         if self.options.double_precision:
             self.cpp_info.defines.append("BOX3D_DOUBLE_PRECISION")
         if self.options.shared and self.settings.compiler == "msvc":

